### PR TITLE
fix: Using Get LUIS or Get QnA Key option does not work in Composer 

### LINKED ISSUE #6963: e2e testing: Using Get LUIS or Get QnA Key option does not work in Composer
## Version
image

## Browser

- [X] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [X] Windows
- [ ] Ubuntu

## To Reproduce

1. Create an Enterprise bot in Composer
2. Click on 'Project Settings' in left nav to go to the following page

image

3. Click on 'Get LUIS keys', the dialog box is greyed out, see below:
image

Note: QnA keys also show a dialog that has the subscription greyed out.

## Expected behavior
I should be able to select the subscription and proceed.

### PR DESCRIPTION
**Pull request by VamsiModem**
## Description
Users with MSA accounts will be able select/create LUIS/QnA resources. 

## Task Item
fixes #6963 

## Screenshots
image
image
image
image

### CODE REVIEW

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 1]
```diff
@@ -489,8 +550,22 @@ export const ManageService = (props: ManageServiceProps) => {
           </p>
           <div css={mainElementStyle}>
             <Dropdown
+              required
+              disabled={allTenants.length === 1 || tenantId != null}
```
[the remark this answers is not in the document]
Comment (hatpick) [reply Composer/packages/client/src/components/ManageService/ManageService.tsx:554]: @GeoffCoxMSFT hidden vs. disabled dropdown, which one should it be?

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 2]
```diff
@@ -695,6 +755,90 @@ export const ManageService = (props: ManageServiceProps) => {
...
+            {props.learnMore ? (
+              <Link href={props.learnMore} target={'_blank'}>
+                {formatMessage('Learn more')}
+              </Link>
+            ) : null}
+          </p>
+          <div css={mainElementStyle}>
+            <Dropdown
+              required
+              disabled={allTenants.length === 1 || tenantId != null}
```
Comment (hatpick) [inline Composer/packages/client/src/components/ManageService/ManageService.tsx:776]: same here

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 3]
```diff
@@ -695,6 +755,90 @@ export const ManageService = (props: ManageServiceProps) => {
...
+              options={allTenants.map((t) => ({ key: t.tenantId, text: t.displayName }))}
+              selectedKey={tenantId}
+              styles={dropdownStyles}
+              onChange={(_e, o) => {
+                setTenantId(o?.key as string);
+              }}
+            />
+            <Dropdown
+              required
+              disabled={!(availableSubscriptions?.length > 0)}
```
Comment (hatpick) [inline Composer/packages/client/src/components/ManageService/ManageService.tsx:788]: nit: more readable, ```availableSubscriptions?.length === 0```

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 4]
```diff
@@ -695,6 +755,90 @@ export const ManageService = (props: ManageServiceProps) => {
...
+              errorMessage={subscriptionsErrorMessage}
+              label={formatMessage('Subscription')}
+              options={
+                availableSubscriptions
+                  ?.filter((p) => p.subscriptionId && p.displayName)
+                  .map((p) => {
+                    return { key: p.subscriptionId ?? '', text: p.displayName ?? 'Unnamed' };
```
Comment (hatpick) [inline Composer/packages/client/src/components/ManageService/ManageService.tsx:795]: unnamed needs to be localized?

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 5]
```diff
@@ -695,6 +755,90 @@ export const ManageService = (props: ManageServiceProps) => {
...
+              selectedKey={subscriptionId}
+              styles={dropdownStyles}
+              onChange={onChangeSubscription}
+            />
+          </div>
+        </div>
+        <DialogFooter>
+          {loading && <Spinner label={loading} labelPosition="right" styles={{ root: { float: 'left' } }} />}
+          <DefaultButton
+            disabled={loading !== undefined}
```
Comment (hatpick) [inline Composer/packages/client/src/components/ManageService/ManageService.tsx:808]: ```disabled={!!loading}```

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 6]
```diff
@@ -695,6 +755,90 @@ export const ManageService = (props: ManageServiceProps) => {
...
+            text={formatMessage('Back')}
+            onClick={() => setCurrentPage(1)}
+          />
+          <PrimaryButton
+            disabled={loading !== undefined || !tenantId || !subscriptionId}
```
Comment (hatpick) [inline Composer/packages/client/src/components/ManageService/ManageService.tsx:813]: !!loading

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 7]
```diff
@@ -695,6 +755,90 @@ export const ManageService = (props: ManageServiceProps) => {
...
+            text={formatMessage('Next')}
+            onClick={() => setCurrentPage(2.1)}
+          />
+          <DefaultButton disabled={loading !== undefined} text={formatMessage('Cancel')} onClick={props.onDismiss} />
```
Comment (hatpick) [inline Composer/packages/client/src/components/ManageService/ManageService.tsx:817]: !!loading

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 8]
```diff
@@ -737,10 +881,7 @@ export const ManageService = (props: ManageServiceProps) => {
         }}
         onDismiss={loading ? () => {} : props.onDismiss}
       >
-        {currentPage === 1 && renderPageOne()}
-        {currentPage === 2 && nextAction === 'choose' && renderPageChoose()}
-        {currentPage === 2 && nextAction === 'create' && renderPageCreate()}
-        {currentPage === 3 && renderPageThree()}
+        {getPage()}
```
Comment (hatpick) [inline Composer/packages/client/src/components/ManageService/ManageService.tsx:884]: renderCurrentStep

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 9]
```diff
@@ -695,6 +755,90 @@ export const ManageService = (props: ManageServiceProps) => {
...
+            onClick={() => setCurrentPage(2.1)}
+          />
+          <DefaultButton disabled={loading !== undefined} text={formatMessage('Cancel')} onClick={props.onDismiss} />
+        </DialogFooter>
+      </div>
+    );
+  };
+
+  const getPage = () => {
+    switch (currentPage as Page) {
```
Comment (hatpick) [inline Composer/packages/client/src/components/ManageService/ManageService.tsx:824]: nit: change currentPage to currentStep

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 10]
```diff
@@ -132,6 +141,69 @@ export const ManageService = (props: ManageServiceProps) => {
...
+  }, []);
+  useEffect(() => {
+    if (tenantId) {
+      AuthClient.getARMTokenForTenant(tenantId)
+        .then((token) => {
+          setToken(token);
+          setTenantsErrorMessage(undefined);
+        })
+        .catch((err) => {
+          setTenantId(undefined);
```
[the remark this answers is not in the document]
Comment (VamsiModem) [reply Composer/packages/client/src/components/ManageService/ManageService.tsx:174]: This won't be necessary, as it won't even execute this  effect as the tenentId is not changed.

File: Composer/packages/client/src/components/ManageService/ManageService.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (VamsiModem) [reply Composer/packages/client/src/components/ManageService/ManageService.tsx]: In the azureProvisionDialog, we are just disabling the dropdown, when there is only one tenant or if it's invalid tenantId.

**Review by hatpick — DISMISSED** [review]

**Review by hatpick — DISMISSED** [review]

**Review by hatpick — DISMISSED** [review]

File: Composer/packages/client/src/components/ManageService/ManageService.tsx [hunk 11]
```diff
@@ -441,13 +500,13 @@ export const ManageService = (props: ManageServiceProps) => {
...
-  const renderPageOne = () => {
+  const renderIntroStep = () => {
     return (
       <div>
         <div css={dialogBodyStyles}>
           <p css={{ marginTop: 0 }}>
             {formatMessage(
-              'Choose from existing {service} keys, create a new {service} resource, or generate a request to handoff to your Azure admin. ',
+              'Select your Azure directory, then choose the subscription where your existing resource is located and the keys you want to use. ',
               { service: props.serviceName }
```
Comment (pavolum) [inline Composer/packages/client/src/components/ManageService/ManageService.tsx:510]: This looks unused with the new string, if so it should be removed.

**Review by pavolum — DISMISSED** [review]
small change needed that has been noted above

**Review by pavolum — APPROVED** [review]

**Review by GeoffCoxMSFT — APPROVED** [review]

**Comment by cwhitten** [comment]
This commit looks to have broken our end to end test suite with the following error:

 Luis Deploy
    (Attempt 1 of 3) can deploy luis success
    (Attempt 2 of 3) can deploy luis success
    1) can deploy luis success


  0 passing (1m)
  1 failing

  1) Luis Deploy
       can deploy luis success:
     TestingLibraryElementError: Timed out retrying: Unable to find an element with the text: westus. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.
      at Object.getElementError (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:918:19)
      at eval (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:3590:38)
      at eval (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:3562:17)
      at eval (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:3606:19)
      at baseCommandImpl (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:700:16)
      at commandImpl (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:703:40)
      at getValue (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:727:23)
      at resolveValue (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:767:35)


I'm curious how this was merged without this successfully passing, unless it was over-ridden by an admin.

**Comment by VamsiModem** [comment]
Hi Chris, For some reason, it didn't block me from checking in, anyhow I have updated the tests, and will push em in, once it's reviewed.

**Comment by cwhitten** [comment]
Thank you Vamsi

**Pull request merged**